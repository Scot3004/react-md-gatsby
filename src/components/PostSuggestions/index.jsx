import React, { Component } from "react";
import FontIcon from "react-md/lib/FontIcons";
import Link from "gatsby-link";
import { translate } from "react-i18next";
import "./PostSuggestions.scss";

class PostSuggestions extends Component {
  render() {
    const { postNode, t } = this.props;
    const postFields = postNode.fields;
    return (
      <div className="post-suggestions md-grid md-cell--12">
        <Link to={postFields.prevSlug} className="post-suggestion">
          <FontIcon
            forceFontSize
            forceSize={64}
            className="secondary-color arrow-nav"
          >
            arrow_back
          </FontIcon>
          <div className="headline-container hide-on-mobile">
            <h2 className="md-body-2 secondary-color">
              {t("commons:previous")}
            </h2>
            <h6 className="md-headline secondary-color">
              {postFields.prevTitle}
            </h6>
          </div>
        </Link>
        <Link to={postFields.nextSlug} className="post-suggestion">
          <div className="headline-container">
            <h2 className="md-body-2 secondary-color">{t("commons:next")}</h2>
            <h6 className="md-headline secondary-color">
              {postFields.nextTitle}
            </h6>
          </div>
          <FontIcon
            forceFontSize
            forceSize={64}
            className="secondary-color arrow-nav"
          >
            arrow_forward
          </FontIcon>
        </Link>
      </div>
    );
  }
}

export default translate("Blog")(PostSuggestions);
