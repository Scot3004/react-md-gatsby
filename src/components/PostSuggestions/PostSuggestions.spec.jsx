import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PostSuggestions from './PostSuggestions';

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  translate: () => Component => {
    // eslint-disable-next-line no-param-reassign
    Component.defaultProps = { ...Component.defaultProps, t: (text) => text };
    return Component;
  },
}));

const postNode = {
  fields: {
    prevSlug: "prevSlug",
    prevTitle: "prevTitle",
    nextSlug: "nextSlug",
    nextTitle: "nextTitle"
  }
}

it('renders correctly', () => {
  const tree = shallow(<PostSuggestions postNode={postNode} />);
  expect(toJson(tree)).toMatchSnapshot();
});
