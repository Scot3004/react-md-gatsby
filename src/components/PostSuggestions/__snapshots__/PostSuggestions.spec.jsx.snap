// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="post-suggestions md-grid md-cell--12"
>
  <GatsbyLink
    className="post-suggestion"
    to="prevSlug"
  >
    <FontIcon
      className="secondary-color arrow-nav"
      forceFontSize={true}
      forceSize={64}
      iconClassName="material-icons"
    >
      arrow_back
    </FontIcon>
    <div
      className="headline-container hide-on-mobile"
    >
      <h2
        className="md-body-2 secondary-color"
      >
        commons:previous
      </h2>
      <h6
        className="md-headline secondary-color"
      >
        prevTitle
      </h6>
    </div>
  </GatsbyLink>
  <GatsbyLink
    className="post-suggestion"
    to="nextSlug"
  >
    <div
      className="headline-container"
    >
      <h2
        className="md-body-2 secondary-color"
      >
        commons:next
      </h2>
      <h6
        className="md-headline secondary-color"
      >
        nextTitle
      </h6>
    </div>
    <FontIcon
      className="secondary-color arrow-nav"
      forceFontSize={true}
      forceSize={64}
      iconClassName="material-icons"
    >
      arrow_forward
    </FontIcon>
  </GatsbyLink>
</div>
`;
